modules/top-level/nixpkgs.nix

```diff
@@ -2,11 +2,34 @@
   config,
   options,
   lib,
+  pkgs,
   ...
 }:
 let
   cfg = config.nixpkgs;
   opt = options.nixpkgs;
+
+  isConfig = x: builtins.isAttrs x || lib.isFunction x;
+
+  mergeConfig =
+    lhs_: rhs_:
+    let
+      optCall = maybeFn: x: if lib.isFunction maybeFn then maybeFn x else maybeFn;
+      lhs = optCall lhs_ { inherit pkgs; };
+      rhs = optCall rhs_ { inherit pkgs; };
+    in
+    lib.recursiveUpdate lhs rhs
+    // lib.optionalAttrs (lhs ? packageOverrides) {
+      packageOverrides =
+        pkgs:
+        optCall lhs.packageOverrides pkgs // optCall (lib.attrByPath [ "packageOverrides" ] { } rhs) pkgs;
+    }
+    // lib.optionalAttrs (lhs ? perlPackageOverrides) {
+      perlPackageOverrides =
+        pkgs:
+        optCall lhs.perlPackageOverrides pkgs
+        // optCall (lib.attrByPath [ "perlPackageOverrides" ] { } rhs) pkgs;
+    };
```

Comment:
I have not found the documentation for this override in the nixpkgs manual, you got it from upstream's config type? Should we leave it?

Reply:
Yeah, this is taken from upstream: https://github.com/NixOS/nixpkgs/blob/c7f73ed1ad47009a95815327d478dba0db03c844/nixos/modules/misc/nixpkgs.nix#L16-L33

I assume it is there to cover some edge cases when merging two `nixpkgs.config` definitions. I.e. one def having a `perlPackageOverrides` _attrset_ and the other having a `perlPackageOverrides` _function_.